pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/PatchRoute.java

```diff
@@ -65,11 +65,11 @@ public Object processRequest(Request request, Response response, String userId)
                 throw new AuthorizationException("User is not authorized to patch participant data");
             }
         } catch (JsonSyntaxException e) {
-            throw new DSMBadRequestException("Invalid request payload format for patch");
+            throw new DSMBadRequestException("Invalid request payload format for patch", e);
         } catch (DuplicateException e) {
             // TODO: fix this by inspecting the DuplicateException throwers to see if cases are due to
             // bad request data (400)
-            throw new DsmInternalError("Duplicate value", e);
+            throw new DsmInternalError("Duplicate value:" + e.getMessage(), e);
```

Comment:
As my comment above mentions, ideally we convert the DuplicateException in Patch.java to a DsmInternalError, thus we don't have to catch it here (at least that particular instance). That allows us to avoid this particular fancy footwork.
DuplicateException is not buying us much here (or elsewhere), since we can (and do) describe the particular error in the exception message. What we want to be sure of is whether this is an internal error (and thus mapped to 500) or an issue with a user input. In the latter case we want to use DSMBadRequestException, which gets mapped to 400.
Since you are already working around the problems with DuplicateException, perhaps now is the right time to make the change in Patch.java? (Which would also match our general error handling pattern in SQL query code.)

Reply:
I've replaced `DuplicateException`, but I don't think now is the right time to make decisions about what is a 400.  I'm in favor of getting some road time with this and seeing if the additional logging can help us determine what should be a 400.